examples/querying_entity_versions.ipynb

```diff
@@ -0,0 +1,782 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "0fcf0214-2561-488a-8eee-0330ca27710d",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from openassetio_mediacreation.traits.lifecycle import VersionTrait\n",
+    "from openassetio.access import ResolveAccess\n",
+    "\n",
+    "entity_data = manager.resolve(logo_ref, {VersionTrait.kId}, ResolveAccess.kRead, context)\n",
+    "\n",
+    "version_trait = VersionTrait(entity_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a6c4f0ff-0e5d-40b7-84d6-d46efa8cba7a",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "We use the `VersionTrait` class as a wrapper around the generic `TraitsData` returned from the API to inspect it's properties - which contain information about the entity's version. This helps ensure we don't accidentally mis-spell traits or their properties.\n",
+    "\n",
+    "MediaCreation use the term \"tag\" for some string that identifies the version of an entity. Using a string, rather than a number allows full flexibility in how a specific manager may wish to handle versions and dynamic \"meta-versions\".\n",
+    "\n",
+    "**Note: that the tags themselves are specific to each manager, and are not defined by the API or the MediaCreation library. Hosts should not attempt to string munge entity references, and use the methods covered in this document instead if they wish to obtain references for specific versions. Otherwise, this defeats the function of the API as an abstraction layer.**\n",
+    "\n",
+    "The `specifiedTag` property holds the version tag addressed by the entity reference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "1c0683dd-c601-41b4-a069-d5acb3c40cbf",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "> **Result:**\n",
+       "> `latest`"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "helpers.display_result(version_trait.getSpecifiedTag())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17049575-fc94-4ac1-9db4-466cccbe1061",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "BAL uses the `v=` query param to define versions in its references. Ours did not have one (`\"bal:///project_artwork/logos/openassetio\"`), in this case, BAL elects to return the latest version. Again note, this is a BAL specific behaviour, other managers may handle things differently.\n",
+    "\n",
+    "The version trait defines that the `specifiedTag` property should hold the tag string that could be used to achieve the same result, so is set to `\"latest\"`."
```

Comment:
```suggestion
    "The version trait defines that the `specifiedTag` property will hold the tag string that could be used to achieve the same result, so is set to `\"latest\"`."
```
[consider][minor] Getting super nitpicky now, but this mirrors the verbiage used in the docs, and I think it's the right choice as should introduced unnecessary doubt. 

Getting very technical writer here, but I'd love to have consistent principles governing our shoulds/wills/musts, alas if only we had all the time in the world.

Reply:
As the value is optional, does that affect things at all?

Reply:
I see what you mean, it might, I'm no expert. If we were to be more explicit on that point it might become something like :

"The version trait defines that, as the `specifiedTag` property does not contain a version specifier, it should hold the tag string that could be used to achieve the same result, so is set to `\"latest\"`."

Although even then, `will` and `should` are practically equivilent so we should default to what's used in the trait documentation. Dosen't matter too much though so go with your gut.

Reply:
Good call, how about [this](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/59/commits/8c499276c1c9901ba9c9b0763bb7ebb5b6579703) - attempted to highlight the BAL specifics of it to a greater extent. (nb, further updated in response to the comment below [here](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/59/commits/4b298142bda5caf00b08022170570c9a1c457e79))